src/main/java/uk/gov/digital/ho/hocs/casework/api/ActionDataExternalInterestService.java

```diff
@@ -0,0 +1,158 @@
+
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.*;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.ACTION_DATA_UPDATE_FAILURE;
+
+@Service
+@Slf4j
+public class ActionDataExternalInterestService implements ActionService {
+    private final ActionDataExternalInterestRepository actionDataExternalInterestRepository;
+    private final CaseDataRepository caseDataRepository;
+    private final InfoClient infoClient;
+    private final AuditClient auditClient;
+    private final CaseNoteService caseNoteService;
+
+    private static final String CREATE_CASE_NOTE_KEY = "RECORD_INTEREST";
+    private static final String UPDATE_CASE_NOTE_KEY = "UPDATE_INTEREST";
+
+    @Autowired
+    public ActionDataExternalInterestService(ActionDataExternalInterestRepository actionDataExternalInterestRepository,
+                                             CaseDataRepository caseDataRepository,
+                                             InfoClient infoClient,
+                                             AuditClient auditClient,
+                                             CaseNoteService caseNoteService) {
+        this.actionDataExternalInterestRepository = actionDataExternalInterestRepository;
+        this.caseDataRepository = caseDataRepository;
+        this.infoClient = infoClient;
+        this.auditClient = auditClient;
+        this.caseNoteService = caseNoteService;
+    }
+
+    @Override
+    public String getServiceDtoTypeKey() {
+        return ActionDataExternalInterestInboundDto.class.getSimpleName();
+    }
+
+    @Override
+    public String getServiceMapKey() {
+        return "RECORD_INTEREST";
+    }
+
+    @Override
+    public void create(UUID caseUuid, UUID stageUuid, ActionDataDto actionData) {
+
+        ActionDataExternalInterestInboundDto actionDataExternalInterestDto =
+                (ActionDataExternalInterestInboundDto) actionData;
+
+        log.debug("Received request to create external " +
+                "interest: {} for case: {}, stage: {}", actionDataExternalInterestDto, caseUuid, stageUuid);
+
+
+        CaseData caseData = caseDataRepository.findActiveByUuid(caseUuid);
+        if (caseData == null) {
+            // Should have exited from the getCase call if no case with ID, however put here for safety to stop orphaned records.
+            throw new ApplicationExceptions.EntityNotFoundException(String.format("Case with id: %s does not exist.", caseUuid), CASE_NOT_FOUND);
+        }
+
+
+        ActionDataExternalInterest actionDataExternalInterest = new ActionDataExternalInterest(
+                actionData.getCaseTypeActionUuid(),
+                actionData.getCaseTypeActionLabel(),
+                caseData.getType(),
+                caseUuid,
+                actionDataExternalInterestDto.getInterestedPartyType(),
+                actionDataExternalInterestDto.getDetailsOfInterest()
+        );
+
+
+        caseDataRepository.save(caseData);
```

Comment:
you don't need this it seems as you're not manipulating the case Data in anyway.